logging/src/lib.rs

```diff
@@ -168,6 +168,25 @@ impl<T, E: Clone> Logger<T, E> {
 }
 
 impl<T, E: Clone, A: ?Sized + FnMut(&Duration, &mut Vec<(Duration, E, T)>)> LoggerInner<T, E, A> {
+
+    /// The upper limit for buffers to allocate, size in bytes. [Self::buffer_capacity] converts
+    /// this to size in elements.
+    const BUFFER_SIZE_BYTES: usize = 1 << 13;
+
+    /// The maximum buffer capacity in elements. Returns a number between [Self::BUFFER_SIZE_BYTES]
+    /// and 1, inclusively.
+    // TODO: This fn is not const because it cannot depend on non-Sized generic parameters
+    fn buffer_capacity() -> usize {
+        let size =  ::std::mem::size_of::<(Duration, E, T)>();
+        if size == 0 {
+            Self::BUFFER_SIZE_BYTES
+        } else if size <= Self::BUFFER_SIZE_BYTES {
+            Self::BUFFER_SIZE_BYTES / size
+        } else {
+            1
+        }
```

Comment:
very minor nit, but is there a reason to do this as if statements, and in the timely crate as a match? This version is clearer to me (unfamiliar with Rust's match guarding logic), but if there is a reason let me know! Can fix or not fix; very minor.